File: mediator/java/ChatUser.java
package Mediator;

// Colleague (User) soyutundan türer.
// Mediator sayesinde diğer sınıf örnekler ile iletişim kurar.
// UML diyagramındaki ConcreteColleague yapısına denk gelmektedir.
class ChatUser extends User {
    public ChatUser(IChatRoomMediator chatRoomMediator) {
        super(chatRoomMediator);
    }
}

File: mediator/kotlin/ChatUser.kt
package Mediator

// Colleague (User) soyutundan türer.
// Mediator sayesinde diğer sınıf örnekler ile iletişim kurar.
// UML diyagramındaki ConcreteColleague yapısına denk gelmektedir.
internal class ChatUser(chatRoomMediator: IChatRoomMediator) : User(chatRoomMediator)
